Example/PaymentSheet Example/PaymentSheet Example/EndpointSelectorViewController.swift

```diff
@@ -0,0 +1,190 @@
+        let cell = UITableViewCell(style: .default, reuseIdentifier: "endpoint")
+        guard let endpoint = endpointFor(indexPath: indexPath) else {
+            return cell
+        }
+        cell.textLabel?.text = endpoint
+        cell.textLabel?.font = .preferredFont(forTextStyle: .caption1)
+        cell.textLabel?.numberOfLines = 0
+        return cell
+    }
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard let endpoint = endpointFor(indexPath: indexPath) else {
+            return
+        }
+        self.selectorDelegate?.selected(endpoint: endpoint)
+        self.navigationController?.popViewController(animated: true)
+    }
+}
+// MARK: Cheap Data provider
+extension EndpointSelectorViewController {
+    func loadEndpointSelector(endpoint: String) {
+        let request = URLRequest(url: URL(string: endpoint)!)
+        let session = URLSession.shared.dataTask(with: request) { data, response, error in
+            guard error == nil,
+                  let data = data,
+                  let specs = self.deserializeResponse(data: data) else {
+                fatalError("FAILED on endpoint selector url request")
+                return
+            }
+            self.endpointSpecs = specs
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
+        session.resume()
+    }
+    func deserializeResponse(data: Data) -> EndpointSpec? {
+        let decoder = JSONDecoder()
+        do {
+            return try decoder.decode(EndpointSpec.self, from: data)
+        } catch {
+            return nil
+        }
+    }
+}
+
+// MARK: - Helpers
+extension EndpointSelectorViewController {
+    func endpointFor(indexPath: IndexPath) -> String? {
+        if indexPath.section == 0 {
+            return currentCheckoutEndpoint
+        }
+        guard let specs = endpointSpecs,
+              let sortedSection = sortedSections() else {
+            return nil
+        }
+        let sectionKey = sortedSection[indexPath.section-1].key
+        guard let endpointsInSection = specs.endpointMap[sectionKey] else {
+            return nil
+        }
+        return endpointsInSection[indexPath.row]
+    }
+
+    func sortedSections() -> [Dictionary<String, [String]>.Element]? {
+        guard let specs = endpointSpecs else {
+            return nil
+        }
+        return specs.endpointMap.sorted { k1, k2 in
+            k1.key < k2.key
+        }
+    }
+}
+
+struct EndpointSpec: Codable {
+    let endpointMap: [String: [String]]
+
+    private enum CodingKeys: String, CodingKey {
+        case endpointMap = "endpoint_map"
+    }
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.endpointMap = try container.decode([String: [String]].self, forKey: .endpointMap)
+    }
```

Comment:
You shouldn't need all these if you use `KeyDecodingStrategy.convertFromSnakeCase`